From 89d179458c7f4d4a1f8430558c2b8f8263679ccf Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@suse.com>
Date: Mon, 11 Apr 2016 21:26:06 +0200
Subject: [PATCH] Disable calamari role in Crowbar for appliance

The appliance doesn't ship calamari, due to conflicting django
requirement with horizon.
---
 chef/data_bags/crowbar/template-ceph.json                            | 1 -
 crowbar_framework/app/models/ceph_service.rb                         | 2 +-
 crowbar_framework/app/views/barclamp/ceph/_edit_attributes.html.haml | 2 +-
 3 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/chef/data_bags/crowbar/template-ceph.json b/chef/data_bags/crowbar/template-ceph.json
index cf66b46..2f71f2a 100644
--- a/chef/data_bags/crowbar/template-ceph.json
+++ b/chef/data_bags/crowbar/template-ceph.json
@@ -63,7 +63,6 @@
       },
       "elements": {},
       "element_order": [
-        [ "ceph-calamari" ],
         [ "ceph-mon" ],
         [ "ceph-osd" ],
         [ "ceph-radosgw" ],
diff --git a/crowbar_framework/app/models/ceph_service.rb b/crowbar_framework/app/models/ceph_service.rb
index 8d447fe..06079d7 100644
--- a/crowbar_framework/app/models/ceph_service.rb
+++ b/crowbar_framework/app/models/ceph_service.rb
@@ -144,7 +144,7 @@ class CephService < PacemakerServiceObject
     calamari_node = calamari_nodes.first
 
     base["deployment"]["ceph"]["elements"] = {
-        "ceph-calamari" => calamari_node.nil? ? [] : [calamari_node.name],
+        #"ceph-calamari" => calamari_node.nil? ? [] : [calamari_node.name],
         "ceph-mon" => mon_nodes.map { |x| x.name },
         "ceph-osd" => osd_nodes.map { |x| x.name },
         "ceph-mds" => mds_node.nil? ? [] : [mds_node.name],
diff --git a/crowbar_framework/app/views/barclamp/ceph/_edit_attributes.html.haml b/crowbar_framework/app/views/barclamp/ceph/_edit_attributes.html.haml
index bedde22..282c7f0 100644
--- a/crowbar_framework/app/views/barclamp/ceph/_edit_attributes.html.haml
+++ b/crowbar_framework/app/views/barclamp/ceph/_edit_attributes.html.haml
@@ -23,7 +23,7 @@
         = string_field %w(radosgw ssl keyfile)
         = boolean_field %w(radosgw ssl insecure)
 
-    %fieldset
+    -#%fieldset
       %legend
         = t(".calamari.header")
 
-- 
2.10.1

